### ISSUE TITLE
Filter() return type in summarize_sample_counts.py

### ISSUE THREAD

**Issue by danielm710** [jmtsuji/qiime2-helpers#3]
## Problem
It looks like the current version of the script assumes filter() returns a list (which is a fair assumption in Python 2). I am not sure which Python version you are using, but in Python 3.5 (and probably all versions of Python 3), filter() returns a Filter object, not a list. 

```
biom_filepath = filter(lambda x: BIOM_PATH_PARTIAL in x, qza_data_contents) # line 48
# TODO - check this is a list of length 1
logger.debug("Found biom file in QZA file at " + str(biom_filepath[0]))

# Extract the biom file to /tmp
tmpdir = os.path.join('/tmp', uuid.uuid4().hex)
extraction_path = qza_data.extract(biom_filepath[0], path = tmpdir) 
logger.debug("Extracted temp BIOM file to " + extraction_path) # line 55
```

This results in `TypeError: 'filter' object is not subscriptable`.

## Suggestions
Convert filter object to list like so?

`biom_filepath = list(filter(lambda x: BIOM_PATH_PARTIAL in x, qza_data_contents))# line 48`

**Comment by jmtsuji** [comment]
@danielm710 I've addressed this in commit fb3308a on the develop branch. Can you test and confirm it works?

I think I coded this originally in Python2 because [the conda install of `biom-format`](https://anaconda.org/bioconda/biom-format) was only Python2 compatible. Not sure if this is still the case. But I'm happy to make the script Python3 compatible.

**Comment by danielm710** [comment]
Works in Python3. However, when I tried running the script in Python2, it gave me a different error. I'll open a new issue if it seems to be a bug.
